Tests/Algorithm/AlgorithmSettingsTest.cs

```diff
@@ -77,6 +79,71 @@ public void DefaultValueOfSetHoldingsBufferWorksCorrectly()
             Assert.AreEqual(4986m, actual);
         }
 
+        [TestCase(BrokerageName.FTX, 365)]
+        [TestCase(BrokerageName.RBI, 252)]
+        [TestCase(BrokerageName.Eze, 252)]
+        [TestCase(BrokerageName.Axos, 252)]
+        [TestCase(BrokerageName.Samco, 252)]
+        [TestCase(BrokerageName.FTXUS, 365)]
+        [TestCase(BrokerageName.Bybit, 365)]
+        [TestCase(BrokerageName.Kraken, 365)]
+        [TestCase(BrokerageName.Exante, 252)]
+        [TestCase(BrokerageName.Binance, 365)]
+        [TestCase(BrokerageName.Default, 252)]
+        [TestCase(BrokerageName.Zerodha, 252)]
+        [TestCase(BrokerageName.Bitfinex, 365)]
+        [TestCase(BrokerageName.Wolverine, 252)]
+        [TestCase(BrokerageName.TDAmeritrade, 252)]
+        [TestCase(BrokerageName.FxcmBrokerage, 252)]
+        [TestCase(BrokerageName.OandaBrokerage, 252)]
+        [TestCase(BrokerageName.BinanceFutures, 365)]
+        [TestCase(BrokerageName.TradierBrokerage, 252)]
+        [TestCase(BrokerageName.BinanceCoinFutures, 365)]
+        [TestCase(BrokerageName.TradingTechnologies, 252)]
+        [TestCase(BrokerageName.QuantConnectBrokerage, 252)]
+        [TestCase(BrokerageName.Coinbase, 365, AccountType.Cash)]
+        [TestCase(BrokerageName.BinanceUS, 365, AccountType.Cash)]
+        [TestCase(BrokerageName.InteractiveBrokersBrokerage, 252)]
+        public void ReturnUniqueTradingDayPerYearDependOnBrokerageName(BrokerageName brokerageName, int expectedTradingDayPerYear, AccountType accountType = AccountType.Margin)
+        {
+            var algorithm = new QCAlgorithm();
+            algorithm.SetBrokerageModel(brokerageName, accountType);
+
+            BaseSetupHandler.SetBrokerageTradingDayPerYear(algorithm);
+
+            Assert.AreEqual(expectedTradingDayPerYear, algorithm.Settings.TradingDaysPerYear.Value);
+        }
+
+        [TestCase(BrokerageName.Bybit, 202, 365)]
+        [TestCase(BrokerageName.InteractiveBrokersBrokerage, 404, 252)]
+        public void ReturnCustomTradingDayPerYearIndependentlyFromBrokerageName(BrokerageName brokerageName, int customTradingDayPerYear, int expectedDefaultTradingDayPerYearForBrokerage)
+        {
+            var algorithm = new QCAlgorithm();
+            algorithm.SetBrokerageModel(brokerageName);
+            algorithm.Settings.TradingDaysPerYear = customTradingDayPerYear;
+
+            // duplicate: make sure that custom value is assigned
+            BaseSetupHandler.SetBrokerageTradingDayPerYear(algorithm);
+
+            Assert.AreNotEqual(expectedDefaultTradingDayPerYearForBrokerage, algorithm.Settings.TradingDaysPerYear);
+        }
+
+        [TestCase(null, 252)]
+        [TestCase(404, 404)]
+        public void ReturnTradingDayPerYearWithoutSetBrokerage(int? customTradingDayPerYear = null, int expectedTradingDayPerYear = 0)
```

Comment:
`expectedTradingDayPerYear` should not have a default value, could put this parameter first